plugins/SegmentEditor/tests/Integration/SegmentQueryDecoratorTest.php

```diff
@@ -0,0 +1,78 @@
+<?php
+/**
+ * Piwik - free/libre analytics platform
+ *
+ * @link http://piwik.org
+ * @license http://www.gnu.org/licenses/gpl-3.0.html GPL v3 or later
+ */
+
+namespace Piwik\Plugins\SegmentEditor\tests\Integration;
+
+use Piwik\Plugins\SegmentEditor\API;
+use Piwik\Plugins\SegmentEditor\SegmentQueryDecorator;
+use Piwik\Segment;
+use Piwik\Tests\Framework\Fixture;
+use Piwik\Tests\Framework\TestCase\IntegrationTestCase;
+
+/**
+ * @group SegmentEditor
+ * @group SegmentEditor_Integration
+ */
+class SegmentQueryDecoratorTest extends IntegrationTestCase
+{
+    /**
+     * @var SegmentQueryDecorator
+     */
+    private $segmentQueryDecorator;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        Fixture::createWebsite('2011-01-01');
+        Fixture::createWebsite('2011-01-01');
+        Fixture::createWebsite('2011-01-01');
+
+        $this->segmentQueryDecorator = self::$fixture->piwikEnvironment->getContainer()->get(
+            'Piwik\Plugins\SegmentEditor\SegmentQueryDecorator');
+
+        /** @var API $segmentEditorApi */
+        $segmentEditorApi = self::$fixture->piwikEnvironment->getContainer()->get(
+            'Piwik\Plugins\SegmentEditor\API');
+        $segmentEditorApi->add('segment 1', 'visitCount<2', $idSite = false, $autoArchive = true);
+        $segmentEditorApi->add('segment 2', 'countryCode==fr', $idSite = false, $autoArchive = true);
+        $segmentEditorApi->add('segment 3', 'visitCount<2', 1, $autoArchive = true);
+        $segmentEditorApi->add('segment 4', 'visitCount<2', 2, $autoArchive = true);
+
+        // test that segments w/ auto archive == false aren't included
```

Comment:
segments w/ auto archive === false should also be included, since it's useful to know those segment IDs...

optional:  display `trigger = archive` in the comment when `core:archive` triggered the query